diff -Nur --exclude *~ iproute2/include/rt_names.h iproute2.new/include/rt_names.h
--- iproute2/include/rt_names.h	Sun Apr 16 19:42:50 2000
+++ iproute2.new/include/rt_names.h	Sat Apr 14 17:13:33 2001
@@ -1,6 +1,7 @@
 #ifndef RT_NAMES_H_
 #define RT_NAMES_H_ 1
 
+#include <asm/types.h>
 const char* rtnl_rtprot_n2a(int id, char *buf, int len);
 const char* rtnl_rtscope_n2a(int id, char *buf, int len);
 const char* rtnl_rttable_n2a(int id, char *buf, int len);
