diff -urN iproute2-2.6.7.org/misc/arpd.c iproute2-2.6.7/misc/arpd.c
--- iproute2-2.6.7.org/misc/arpd.c	2004-06-16 21:07:09.442520088 +0200
+++ iproute2-2.6.7/misc/arpd.c	2004-06-16 23:47:31.283777880 +0200
@@ -16,7 +16,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <netdb.h>
-#include <db41/db_185.h>
+#include <db_185.h>
 #include <sys/ioctl.h>
 #include <sys/poll.h>
 #include <errno.h>
diff -urN iproute2-2.6.7.org/misc/Makefile iproute2-2.6.7/misc/Makefile
--- iproute2-2.6.7.org/misc/Makefile	2004-06-16 21:07:09.441520240 +0200
+++ iproute2-2.6.7/misc/Makefile	2004-06-16 23:48:02.212076064 +0200
@@ -22,7 +22,7 @@
 	$(CC) $(CFLAGS) $(LDFLAGS) -o rtacct $(RTACCTOBJ) $(LIBNETLINK) -lm
 
 arpd: $(ARPDOBJ)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o arpd $(ARPDOBJ) $(LIBNETLINK) -ldb-4.1
+	$(CC) $(CFLAGS) $(LDFLAGS) -o arpd $(ARPDOBJ) $(LIBNETLINK) -ldb
 
 rtstat: $(RTSTATOBJ)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o rtstat $(RTSTATOBJ)
