diff -ur iproute2-2.6.18-061002.org/netem/Makefile iproute2-2.6.18-061002/netem/Makefile
--- iproute2-2.6.18-061002.org/netem/Makefile	2006-10-02 22:13:34.000000000 +0200
+++ iproute2-2.6.18-061002/netem/Makefile	2006-10-03 09:00:45.533018231 +0200
@@ -7,7 +7,7 @@
 all: $(DISTGEN) $(DISTDATA)
 
 $(DISTGEN):
-	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
+	$(HOSTCC) $(LDFLAGS) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 %.dist: %
 	./$* > $@

