--- iproute2-4.17.0.org/ip/iproute.c	2010-08-04 17:45:59.000000000 +0000
+++ iproute2-4.17.0/ip/iproute.c	2010-08-25 12:57:40.000000000 +0000
@@ -160,7 +160,7 @@
 	if (r->rtm_family == AF_INET6 && table != RT_TABLE_MAIN)
 		ip6_multiple_tables = 1;
 
-	if (filter.cloned == !(r->rtm_flags & RTM_F_CLONED))
+	if (filter.cloned && !(r->rtm_flags & RTM_F_CLONED))
 		return 0;
 
 	if (r->rtm_family == AF_INET6 && !ip6_multiple_tables) {
