diff -urN iproute2-2.6.8.org/Makefile iproute2-2.6.8/Makefile
--- iproute2-2.6.8.org/Makefile	2004-07-16 21:49:23.066528064 +0200
+++ iproute2-2.6.8/Makefile	2004-07-16 21:50:01.491686552 +0200
@@ -10,7 +10,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 #options for decnet
@@ -21,7 +21,7 @@
 
 CC = gcc
 HOSTCC = gcc
-CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall
+CCOPTS = -D_GNU_SOURCE $(OPT) -Wstrict-prototypes -Wall
 CFLAGS = $(CCOPTS) -I../include $(DEFINES)
 
 LDLIBS += -L../lib -lnetlink -lutil
--- iproute2-050124/ip/iptunnel.c.orig	2005-01-25 01:13:46.000000000 +0100
+++ iproute2-050124/ip/iptunnel.c	2005-01-25 10:26:02.716547608 +0100
@@ -16,6 +16,7 @@
  * Phil Karn <karn@ka9q.ampr.org>	990408:	"pmtudisc" flag
  */
 
+#include <asm/byteorder.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
