--- iproute2-2.6.8.org/Makefile	2004-07-16 21:49:23.066528064 +0200
+++ iproute2-2.6.8/Makefile	2004-07-16 21:50:01.491686552 +0200
@@ -10,7 +10,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 #options for decnet
@@ -21,7 +21,7 @@
 
 CC = gcc
 HOSTCC = gcc
-CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall
+CCOPTS = -D_GNU_SOURCE $(OPT) -Wstrict-prototypes -Wall
 CFLAGS = $(CCOPTS) -I../include $(DEFINES)
 
 LDLIBS += -L../lib -lnetlink -lutil
--- iproute2-050124/ip/iptunnel.c.orig	2005-01-25 01:13:46.000000000 +0100
+++ iproute2-050124/ip/iptunnel.c	2005-01-25 10:26:02.716547608 +0100
@@ -16,6 +16,7 @@
  * Phil Karn <karn@ka9q.ampr.org>	990408:	"pmtudisc" flag
  */
 
+#include <asm/byteorder.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- iproute2-2.6.11-050330/configure.orig	2005-04-01 21:58:11.000000000 +0200
+++ iproute2-2.6.11-050330/configure	2005-06-19 22:38:36.000000000 +0200
@@ -8,7 +8,7 @@
 echo "TC schedulers"
 
 echo -n " ATM	"
-cat >/tmp/atmtest.c <<EOF
+cat >atmtest.c <<EOF
 #include <atm.h>
 int main(int argc, char **argv) {
 	struct atm_qos qos;
@@ -16,7 +16,7 @@
 	return 0;
 }
 EOF
-gcc -I$INCLUDE -o /tmp/atmtest /tmp/atmtest.c -latm >/dev/null 2>&1 
+gcc -I$INCLUDE -o atmtest atmtest.c -latm >/dev/null 2>&1 
 if [ $? -eq 0 ]
 then
     echo "TC_CONFIG_ATM:=y" >>Config
@@ -24,4 +24,4 @@
 else
     echo no
 fi
-rm -f /tmp/atmtest.c /tmp/atmtest
+rm -f atmtest*
