--- iproute2/misc/Makefile.orig	2002-01-16 00:30:32.000000000 +0100
+++ iproute2/misc/Makefile	2004-04-04 22:23:44.718120960 +0200
@@ -6,7 +6,7 @@
 RTSTATOBJ=rtstat.o
 
 ALLOBJ=$(SSOBJ) $(NSTATOBJ) $(IFSTATOBJ) $(RTACCTOBJ) $(ARPDOBJ) $(RTSTATOBJ)
-TARGETS=ss nstat ifstat rtacct arpd rtstat
+TARGETS=ss nstat ifstat rtacct rtstat
 
 all: $(TARGETS)
 
--- iproute2/Makefile~	2004-05-01 21:30:53.000000000 +0000
+++ iproute2/Makefile	2004-05-01 21:33:14.000000000 +0000
@@ -8,7 +8,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 #options if you compile with libc5, and without a bind>=4.9.4 libresolv
@@ -24,7 +24,7 @@
 
 
 CC = gcc
-CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g
+CCOPTS = -D_GNU_SOURCE $(OPT) -Wstrict-prototypes -Wall -g
 CFLAGS = $(CCOPTS) -I../include $(DEFINES)
 
 LDLIBS += -L../lib -lnetlink -lutil
