--- iproute2-2.6.38/tc/m_xt.c.org	2011-05-29 10:01:12.291649868 +0200
+++ iproute2-2.6.38/tc/m_xt.c	2011-05-29 10:04:25.340555276 +0200
@@ -162,7 +162,11 @@
 					return -1;
 				}
 				tcipt_globals.opts =
-				    xtables_merge_options(tcipt_globals.opts,
+				    xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+				                          tcipt_globals.orig_opts,
+#endif
+				                          tcipt_globals.opts,
 				                          m->extra_opts,
 				                          &m->option_offset);
 			} else {
@@ -307,7 +311,11 @@
 			}
 
 			tcipt_globals.opts =
-			    xtables_merge_options(tcipt_globals.opts,
+			    xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+			                          tcipt_globals.orig_opts,
+#endif
+			                          tcipt_globals.opts,
 			                          m->extra_opts,
 			                          &m->option_offset);
 		} else {
