diff -ur iproute2-2.6.18-061002.org/genl/Makefile iproute2-2.6.18-061002/genl/Makefile
--- iproute2-2.6.18-061002.org/genl/Makefile	2006-10-02 22:13:34.000000000 +0200
+++ iproute2-2.6.18-061002/genl/Makefile	2006-10-03 09:09:13.940918661 +0200
@@ -9,11 +9,10 @@
 
 GENLLIB :=
 
-LDFLAGS += -Wl,-export-dynamic
-LDLIBS  += -lm -ldl
 
 all: genl
 
 genl: $(GENLOBJ) $(LIBNETLINK) $(LIBUTIL) $(GENLLIB)
+	$(LD) $(LDFLAGS) $(CFLAGS) -Wl,-export-dynamic -o $@ $^ $(LDLIBS) -lm -ldl
 
 install: all
 	install -m 0755 -s genl $(DESTDIR)$(SBINDIR)
diff -ur iproute2-2.6.18-061002.org/netem/Makefile iproute2-2.6.18-061002/netem/Makefile
--- iproute2-2.6.18-061002.org/netem/Makefile	2006-10-02 22:13:34.000000000 +0200
+++ iproute2-2.6.18-061002/netem/Makefile	2006-10-03 09:00:45.533018231 +0200
@@ -7,7 +7,7 @@
 all: $(DISTGEN) $(DISTDATA)
 
 $(DISTGEN):
-	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
+	$(HOSTCC) $(LDFLAGS) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 %.dist: %
 	./$* > $@
diff -ur iproute2-2.6.18-061002.org/tc/Makefile iproute2-2.6.18-061002/tc/Makefile
--- iproute2-2.6.18-061002.org/tc/Makefile	2006-10-02 22:13:34.000000000 +0200
+++ iproute2-2.6.18-061002/tc/Makefile	2006-10-03 09:00:45.533018231 +0200
@@ -52,18 +52,17 @@
 
 LDLIBS += -L. -ltc -lm -ldl
 
-LDFLAGS += -Wl,-export-dynamic
-
 YACC := bison
 LEX := flex
 
 %.so: %.c
-	$(CC) $(CFLAGS) -shared -fpic $< -o $@
+	$(CC) $(LDFLAGS) $(CFLAGS) -shared -fpic $< -o $@
 
 
 all: libtc.a tc $(TCSO)
 
 tc: $(TCOBJ) $(LIBNETLINK) $(LIBUTIL) $(TCLIB)
+	$(LD) $(LDFLAGS) $(CFLAGS) -Wl,-export-dynamic -o $@ $^ $(LDLIBS)
 
 libtc.a: $(TCLIB)
 	$(AR) rcs $@ $(TCLIB)
@@ -80,7 +79,7 @@
 	rm -f emp_ematch.yacc.output
 
 q_atm.so: q_atm.c
-	$(CC) $(CFLAGS) -shared -fpic -o q_atm.so q_atm.c -latm
+	$(CC) $(LDFLAGS) $(CFLAGS) -shared -fpic -o q_atm.so q_atm.c -latm
 
 %.yacc.c: %.y
 	$(YACC) $(YACCFLAGS) -o $@ $<
