diff -urN iproute2.orig/Makefile iproute2/Makefile
--- iproute2.orig/Makefile	Sun Jul 11 18:49:17 1999
+++ iproute2/Makefile	Sun Jul 11 18:49:31 1999
@@ -27,7 +27,7 @@
 endif
 
 CC=gcc
-CFLAGS=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g $(GLIBCFIX) -I$(KERNEL_INCLUDE) -I../include $(DEFINES)
+CFLAGS=-D_GNU_SOURCE $(OPT) -Wstrict-prototypes -Wall -g $(GLIBCFIX) -I$(KERNEL_INCLUDE) -I../include $(DEFINES)
 
 LDLIBS += -L../lib -lnetlink -lutil
 
