--- iproute2/ip/ipaddress.c.orig	Tue Dec  2 14:13:26 2003
+++ iproute2/ip/ipaddress.c	Tue Dec  2 14:19:45 2003
@@ -20,13 +20,13 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
-#include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
 #include <fnmatch.h>
+#include <linux/netdevice.h>
+#include <linux/if_arp.h>
+#include <linux/sockios.h>
 
 #include "rt_names.h"
 #include "utils.h"
--- iproute2/ip/iptunnel.c.orig	Tue Dec  2 14:13:26 2003
+++ iproute2/ip/iptunnel.c	Tue Dec  2 14:23:37 2003
@@ -24,11 +24,11 @@
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
 #include <netinet/in.h>
 #include <netinet/ip.h>
 #include <arpa/inet.h>
+#include <linux/if.h>
+#include <linux/if_arp.h>
 #include <linux/if_tunnel.h>
 #include <asm/byteorder.h>
 
--- iproute2/ip/ipmaddr.c.orig	Mon Dec 24 23:33:02 2001
+++ iproute2/ip/ipmaddr.c	Tue Dec  2 14:23:59 2003
@@ -17,13 +17,13 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <string.h>
 #include <linux/netdevice.h>
 #include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
-#include <string.h>
 
 #include "rt_names.h"
 #include "utils.h"
--- iproute2/ip/ipmroute.c.orig	Mon Dec 24 22:38:55 2001
+++ iproute2/ip/ipmroute.c	Tue Dec  2 14:24:18 2003
@@ -17,13 +17,13 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <string.h>
 #include <linux/netdevice.h>
 #include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
-#include <string.h>
 
 #include "utils.h"
 
--- iproute2/lib/ll_types.c.orig	Wed Dec 26 00:01:31 2001
+++ iproute2/lib/ll_types.c	Tue Dec  2 14:18:59 2003
@@ -17,12 +17,12 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
-#include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <linux/netdevice.h>
+#include <linux/if_arp.h>
+#include <linux/sockios.h>
 
 char * ll_type_n2a(int type, char *buf, int len)
 {
--- iproute2/lib/ll_proto.c.orig	Wed Dec 26 00:18:35 2001
+++ iproute2/lib/ll_proto.c	Tue Dec  2 14:18:54 2003
@@ -17,12 +17,12 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
-#include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <linux/netdevice.h>
+#include <linux/if_arp.h>
+#include <linux/sockios.h>
 
 #include "utils.h"
 
--- iproute2/lib/ll_addr.c.orig	Sun Apr 16 19:42:52 2000
+++ iproute2/lib/ll_addr.c	Tue Dec  2 14:19:16 2003
@@ -17,12 +17,12 @@
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
-#include <linux/sockios.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <linux/netdevice.h>
+#include <linux/if_arp.h>
+#include <linux/sockios.h>
 
 #include "utils.h"
 
