diff -urNbB iproute2-2.6.14-051107.org/configure iproute2-2.6.14-051107/configure
--- iproute2-2.6.14-051107.org/configure	2005-01-18 00:26:46.000000000 +0100
+++ iproute2-2.6.14-051107/configure	2005-11-10 16:04:15.024922750 +0100
@@ -8,7 +8,7 @@
 echo "TC schedulers"
 
 echo -n " ATM	"
-cat >/tmp/atmtest.c <<EOF
+cat >atmtest.c <<EOF
 #include <atm.h>
 int main(int argc, char **argv) {
 	struct atm_qos qos;
@@ -16,7 +16,7 @@
 	return 0;
 }
 EOF
-gcc -I$INCLUDE -o /tmp/atmtest /tmp/atmtest.c -latm >/dev/null 2>&1 
+gcc -I$INCLUDE -o atmtest atmtest.c -latm >/dev/null 2>&1 
 if [ $? -eq 0 ]
 then
     echo "TC_CONFIG_ATM:=y" >>Config
@@ -24,4 +24,4 @@
 else
     echo no
 fi
-rm -f /tmp/atmtest.c /tmp/atmtest
+rm -f atmtest*
diff -urNbB iproute2-2.6.14-051107.org/Makefile iproute2-2.6.14-051107/Makefile
--- iproute2-2.6.14-051107.org/Makefile	2005-07-09 00:08:47.000000000 +0200
+++ iproute2-2.6.14-051107/Makefile	2005-11-10 16:04:15.024922750 +0100
@@ -10,7 +10,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 #options for decnet
@@ -21,7 +21,7 @@
 
 CC = gcc
 HOSTCC = gcc
-CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall
+CCOPTS = -D_GNU_SOURCE $(OPT) -Wstrict-prototypes -Wall
 CFLAGS = $(CCOPTS) -I../include $(DEFINES)
 YACCFLAGS = -d -t -v
 
