--- ./include/pseudo2_4.h.org	Sun Oct 28 22:58:55 2001
+++ ./include/pseudo2_4.h	Sun Oct 28 23:09:54 2001
@@ -0,0 +1,51 @@
+/* 
+ * These are definitions needed to build iproute2 with kernel 2.2
+ */
+
+#ifndef ARPHRD_IEEE1394
+#define ARPHRD_IEEE1394		24	/* IEEE 1394 IPv4 - RFC 2734	*/
+#endif
+
+#ifndef ARPHRD_HWX25
+#define ARPHRD_HWX25		272	/* Boards with X.25 in firmware	*/
+#endif
+#ifndef ARPHRD_DDCMP
+#define ARPHRD_DDCMP		517	/* Digital's DDCMP protocol	*/
+#endif
+
+#ifndef ARPHRD_RAWHDLC
+#define ARPHRD_RAWHDLC		518	/* Raw HDLC                     */
+#endif
+
+#ifndef ARPHRD_IEEE80211
+#define ARPHRD_IEEE80211	801	/* IEEE 802.11                  */
+#endif
+
+#ifndef ETH_P_PUPAT
+#define ETH_P_PUPAT     	0x0201	/* Xerox PUP Addr Trans packet  */
+#endif
+
+#ifndef ETH_P_IEEEPUP
+#define ETH_P_IEEEPUP   	0x0a00	/* Xerox IEEE802.3 PUP packet */
+#endif
+
+#ifndef ETH_P_IEEEPUPAT
+#define ETH_P_IEEEPUPAT 	0x0a01	/* Xerox IEEE802.3 PUP Addr Trans packet */
+#endif
+
+#ifndef ETH_P_PPP_DISC
+#define ETH_P_PPP_DISC		0x8863	/* PPPoE discovery messages     */
+#endif
+
+#ifndef ETH_P_PPP_SES
+#define ETH_P_PPP_SES		0x8864	/* PPPoE session messages       */
+#endif
+
+#ifndef ETH_P_ECONET
+#define ETH_P_ECONET		0x0018	/* Acorn Econet                 */
+#endif
+
+#ifndef ARPHRD_ATM
+#define ARPHRD_ATM      19              /* ATM                          */
+#endif
+

--- ./lib/ll_types.c.org	Sun Oct 28 23:01:45 2001
+++ ./lib/ll_types.c	Sun Oct 28 23:02:17 2001
@@ -23,6 +23,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include "pseudo2_4.h"
 
 char * ll_type_n2a(int type, char *buf, int len)
 {
--- ./lib/ll_proto.c.org	Sun Oct 28 23:10:32 2001
+++ ./lib/ll_proto.c	Sun Oct 28 23:11:01 2001
@@ -25,7 +25,7 @@
 #include <string.h>
 
 #include "utils.h"
-
+#include "pseudo2_4.h"
 
 #define __PF(f,n) { ETH_P_##f, #n },
 static struct {
