--- iproute2-2.6.11/tc/f_u32.c.org	2005-03-14 23:23:54.000000000 +0100
+++ iproute2-2.6.11/tc/f_u32.c	2005-03-15 14:25:55.159534816 +0100
@@ -22,6 +22,8 @@
 #include <string.h>
 #include <linux/if.h>
 
+# include <asm/types.h>
+
 #include "utils.h"
 #include "tc_util.h"
 
@@ -591,6 +593,13 @@
 	return res;
 }
 
+struct tc_u32_mark
+{
+        unsigned int             val;
+        unsigned int             mask;
+        unsigned int             success;
+}	tc_u32_mark;
+
 static int parse_mark(int *argc_p, char ***argv_p, struct nlmsghdr *n)
 {
 	int res = -1;
